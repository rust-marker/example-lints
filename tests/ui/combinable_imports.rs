#![feature(register_tool)]
#![register_tool(marker)]
#![allow(unused_imports)]

mod example_1 {
    use std::fs::File;
    use std::vec::Vec;

    use std::collections::BTreeMap;
    #[allow(marker::combinable_imports)]
    use std::collections::BTreeSet;
}

mod example_2 {
    use std::{collections::BTreeMap, collections::BTreeSet};
}

mod example_3 {
    use std::{
        collections::{BTreeMap, BTreeSet},
        fs::File,
        vec::Vec,
    };
}

/// False positive, waiting on <https://github.com/rust-marker/marker/issues/26>
mod example_4 {
    pub use std::fs::File;
    use std::vec::Vec;
}

fn main() {
    let _x = 18;
}
