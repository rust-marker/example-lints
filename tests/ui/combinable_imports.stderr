warning: this import can be merged...
 --> $DIR/combinable_imports.rs:7:5
  |
7 |     use std::vec::Vec;
  |     ^^^^^^^^^^^^^^^^^^
  |
note: ...with this one
 --> $DIR/combinable_imports.rs:6:5
  |
6 |     use std::fs::File;
  |     ^^^^^^^^^^^^^^^^^^
  = note: `#[warn(marker::combinable_imports)]` on by default

warning: this import can be merged...
 --> $DIR/combinable_imports.rs:9:5
  |
9 |     use std::collections::BTreeMap;
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
note: ...with this one
 --> $DIR/combinable_imports.rs:6:5
  |
6 |     use std::fs::File;
  |     ^^^^^^^^^^^^^^^^^^

warning: this import can be merged...
  --> $DIR/combinable_imports.rs:15:38
   |
15 |     use std::{collections::BTreeMap, collections::BTreeSet};
   |                                      ^^^^^^^^^^^^^^^^^^^^^
   |
note: ...with this one
  --> $DIR/combinable_imports.rs:15:15
   |
15 |     use std::{collections::BTreeMap, collections::BTreeSet};
   |               ^^^^^^^^^^^^^^^^^^^^^

warning: this import can be merged...
  --> $DIR/combinable_imports.rs:29:5
   |
29 |     use std::vec::Vec;
   |     ^^^^^^^^^^^^^^^^^^
   |
note: ...with this one
  --> $DIR/combinable_imports.rs:28:5
   |
28 |     pub use std::fs::File;
   |     ^^^^^^^^^^^^^^^^^^^^^^

warning: 4 warnings emitted

